File: Souqly-API/Models/Product.cs
using System.Net.Mime;
using System;
using System.Collections.Generic;

namespace Souqly_API.Models
{
    public class Product
    {
        public int Id { get; set; }
        public string ProductName { get; set; }
        public float Weight { get; set; }
        public string Description { get; set; }
        public string Brand { get; set; }
        public DateTime Date { get; set; }
        public string Dimension { get; set; }

        public int? QuantitySold { get; set; } = 0;

        public User Supplier { get; set; }
        public Category Category { get; set; }
        public int CategoryId { get; set; }
        public int SupplierId { get; set; }

        public ICollection<Option> Options { get; set; }
        public ICollection<Image> Images { get; set; }

        // Many To Many Realation
     //   public List<ProductOptionCart> ProductCarts { get; set; }
     //   public virtual ICollection<MarketingProduct> MarketingProducts { get; set; }


    }
}

File: Souqly-SPA/_models/ola/Product.ts
// import { Category } from "./Category";
// import { Image } from "./Image";
// import { MarketingProduct } from "./MarketingProduct";
// import { Option } from "./Option";
// import { ProductOptionCart } from "./ProductOptionCart";
// import { User } from "./user";

import { Option } from "./Option";

// export class Product {
//     constructor(public id?:number, public productName?:string, public weight?:number, public description?:string,
//         public brand?:string, public date?:Date, public dimension?:string, public category?:Category,
//         public categoryId?:number, public options?:Option[], public images?:Image[],
//             public productCarts?:ProductOptionCart[], public marketingProducts?:MarketingProduct[],
//             public user?:User[], public userId?:number){

//     }
// } 

export interface Product {
    id:number;
    productName:string;
    stockIn:number;
    price:number;
    images:string[];
    options: Option[];

}

